overlord/devicestate/devicemgr.go

```diff
@@ -987,11 +1011,6 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, flags sn
 		return nil
 	}
 
-	if release.OnClassic {
```

Comment:
here's the test https://github.com/pedronis/snappy/blob/0a9dcc2bb3a758b0a66a40f83c5617ee77e5eef2/overlord/devicestate/devicemgr_test.go#L1419